### ISSUE TITLE
Use of nanmean breaks dipy for numpy < 1.8

### ISSUE THREAD

**Issue by matthew-brett** [nipy/dipy#600]
See: http://nipy.bic.berkeley.edu/builders/dipy-py2.6-32/builds/509/steps/shell_6/logs/stdio

```
======================================================================
ERROR: dipy.reconst.tests.test_sfm.test_sfm
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/buildslave/bongoslave/dipy-py2_6-32/build/venv/local/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/usr/lib/python2.7/dist-packages/numpy/testing/decorators.py", line 146, in skipper_func
    return f(*args, **kwargs)
  File "/home/buildslave/bongoslave/dipy-py2_6-32/build/venv/local/lib/python2.7/site-packages/dipy/reconst/tests/test_sfm.py", line 29, in test_sfm
    sffit1 = sfmodel.fit(data[0, 0, 0])
  File "/home/buildslave/bongoslave/dipy-py2_6-32/build/venv/local/lib/python2.7/site-packages/dipy/reconst/sfm.py", line 380, in fit
    isotropic = self.isotropic(self.gtab).fit(flat_data)
  File "/home/buildslave/bongoslave/dipy-py2_6-32/build/venv/local/lib/python2.7/site-packages/dipy/reconst/sfm.py", line 148, in fit
    p = np.nanmean(to_fit / self.gtab.bvals[~self.gtab.b0s_mask], -1)
AttributeError: 'module' object has no attribute 'nanmean'

======================================================================
ERROR: dipy.reconst.tests.test_sfm.test_exponential_iso
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/buildslave/bongoslave/dipy-py2_6-32/build/venv/local/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/usr/lib/python2.7/dist-packages/numpy/testing/decorators.py", line 146, in skipper_func
    return f(*args, **kwargs)
  File "/home/buildslave/bongoslave/dipy-py2_6-32/build/venv/local/lib/python2.7/site-packages/dipy/reconst/tests/test_sfm.py", line 143, in test_exponential_iso
    sffit1 = sfmodel.fit(data[0, 0, 0])
  File "/home/buildslave/bongoslave/dipy-py2_6-32/build/venv/local/lib/python2.7/site-packages/dipy/reconst/sfm.py", line 380, in fit
    isotropic = self.isotropic(self.gtab).fit(flat_data)
  File "/home/buildslave/bongoslave/dipy-py2_6-32/build/venv/local/lib/python2.7/site-packages/dipy/reconst/sfm.py", line 148, in fit
    p = np.nanmean(to_fit / self.gtab.bvals[~self.gtab.b0s_mask], -1)
AttributeError: 'module' object has no attribute 'nanmean'
```

Unfortunately travis-ci seems to be doing something nasty to its environment variables at the moment, and is no longer testing our minimum dependencies - so travis-ci did not pick this one up.

Numpy 1.8 seems too high a dependency.

**Comment by arokem** [comment]
Oh - good catch! What's a good way out of this? Should we reimplement
`nanmean` locally?

**Comment by arokem** [comment]
Or use `scipy.stats.nanmean` instead?

**Comment by matthew-brett** [comment]
Sorry to say, but yes, I think a copy of nanmean from recent numpy is the way to go...

**Comment by matthew-brett** [comment]
Oops - yes - maybe that would work.

**Comment by arokem** [comment]
Yeah - I just looked and scipy is deprecating stats.nanmean as of 0.15, in
favor of, you guessed it: np.nanmean

http://docs.scipy.org/doc/scipy-dev/reference/generated/scipy.stats.nanmean.html

**Comment by arokem** [comment]
What is someone decides to use the combination of  <0.18 numpy with >0.15
scipy? :-)

**Comment by matthew-brett** [comment]
Right, maybe:

```
try:
    from numpy import nanmean
except ImportError:
    from scipy.stats import nanmean
```
?

**Comment by arokem** [comment]
https://github.com/nipy/dipy/pull/601

**Comment by matthew-brett** [comment]
Thanks - fixed with #601
